From: Robert Mader <robert.mader@collabora.com>
Date: Sat, 28 Sep 2024 23:57:47 +0200
Subject: [PATCH] media: i2c: imx363: Remove 1920_1080 mode

It currently results in a brownish tint for modes <= 1920x1080 and
breaks the mode in question - 1920x1080 - entirely in libcamera[1].

Using the 4k mode is not optimal either and results in a "zoom" effect
when apps request lower modes. Overall it arguably work betters though
and the image result is less confusing to downstream.

1: 1920x1080 apparently is not a valid mode for SRGGB10 and likely the
right crop size hase to be a bit bigger - to be investigated.

Signed-off-by: Robert Mader <robert.mader@collabora.com>
Signed-off-by: Richard Acayan <mailingradian@gmail.com>
---
 drivers/media/i2c/imx363.c | 39 ---------------------------------------
 1 file changed, 39 deletions(-)

diff --git a/drivers/media/i2c/imx363.c b/drivers/media/i2c/imx363.c
index f6319f1..fc1fdda 100644
--- a/drivers/media/i2c/imx363.c
+++ b/drivers/media/i2c/imx363.c
@@ -422,28 +422,6 @@ static const struct cci_reg_sequence mode_4032x3024_regs[] = {
 	{CCI_REG8(0x0227), 0x00},
 };
 
-static const struct cci_reg_sequence mode_1920_1080_regs[] = {
-	{IMX363_REG_BINNING_MODE, 1},
-	{IMX363_REG_BINNING_TYPE_V, 0x42},
-	// {IMX363_REG_SCALE_MODE, 1},
-	// {IMX363_REG_SCALE_M, 64},
-	{IMX363_REG_X_ADD_STA, 0}, // analog cropping
-	{IMX363_REG_Y_ADD_STA, 0},
-	{IMX363_REG_X_ADD_END, 4031},
-	{IMX363_REG_Y_ADD_END, 3023},
-	{IMX363_REG_X_EVN_INC, 1}, //subsampling
-	{IMX363_REG_X_ODD_INC, 1},
-	{IMX363_REG_Y_EVN_INC, 1},
-	{IMX363_REG_Y_ODD_INC, 1},
-	{IMX363_REG_DIG_CROP_X_OFFSET, 0}, // digital cropping
-	{IMX363_REG_DIG_CROP_Y_OFFSET, 0},
-	{IMX363_REG_DIG_CROP_IMAGE_WIDTH, 1920},
-	{IMX363_REG_DIG_CROP_IMAGE_HEIGHT, 1080},
-	{IMX363_REG_X_OUT_SIZE, 1920},
-	{IMX363_REG_Y_OUT_SIZE, 1080},
-	{IMX363_REG_FRM_LENGTH_LINES, IMX363_VTS_30FPS_HD},
-};
-
 /*
  * The supported formats.
  * This table MUST contain 4 entries per format, to cover the various flip
@@ -568,23 +546,6 @@ static const struct imx363_mode supported_modes[] = {
 			.height = IMX363_PIXEL_ARRAY_HEIGHT,
 		},
 	},
-	{
-		.width = 1920,
-		.height = 1080,
-		.vts_def = IMX363_VTS_30FPS_HD,
-		.vts_min = IMX363_VTS_30FPS_HD,
-		.reg_list = {
-			.num_of_regs = ARRAY_SIZE(mode_1920_1080_regs),
-			.regs = mode_1920_1080_regs,
-		},
-		.link_freq_index = IMX363_LINK_FREQ_640MBPS,
-		.crop = {
-			.left = IMX363_PIXEL_ARRAY_LEFT,
-			.top = IMX363_PIXEL_ARRAY_TOP,
-			.width = 4032,
-			.height = 3024,
-		},
-	},
 };
 
 struct imx363 {
